Test/Tests/RealtimeClientPresenceTests.swift

```diff
@@ -745,86 +745,106 @@ class RealtimeClientPresenceTests: XCTestCase {
         }
     }
 
-    func test__023__Presence__Channel_state_change_side_effects__channel_enters_the_SUSPENDED_state__should_maintain_the_PresenceMap_and_any_members_present_before_and_after_the_sync_should_not_emit_presence_events() throws {
+    func test__023__Presence__Channel_state_change_side_effects__channel_enters_the_SUSPENDED_state__members_map_is_preserved_and_only_members_that_changed_between_ATTACHED_states_should_result_in_presence_events() throws {
```

Comment:
This test fails because of additional UPDATE event in v2, but I didn't want to just omit it like Ikbal did, instead I've decided to rewrite it, because in its current setup it doesn't make a lot of sense. I mean what it should ensure, is that for every member whose state changed during SUSPENDED an appropriate event should be generated after client reconnects:

>This ensures that if the channel later becomes @ATTACHED@, it will only emit presence events for the changes in the @PresenceMap@ that have occurred whilst the client was disconnected.

How [leaving](https://github.com/ably/ably-cocoa/blob/273277e566fc1641cc8fc5b6473999a3e9ab6c4e/Test/Tests/RealtimeClientPresenceTests.swift#L812C37-L812C37) after channel was attached demonstrates that is unclear for me.

What I did is a separate client for the member that will leave during SUSPENDED and ensure that a leave event for this member will be generated upon new attach. Since this setup still makes sense in v1 I've made a separate PR for that - https://github.com/ably/ably-cocoa/pull/1793
It also solves this problem with an UPDATE event, because it will not be generated for the separate client.